examples/ssh/macos_applicaton.js

```diff
@@ -0,0 +1,204 @@
+/**
+ * Domotz Custom Driver
+ * Name: macOS - Applications
+ * Description: Retrieves details of installed applications on macOS
+ *
+ * Communication protocol is SSH
+ *
+ * Tested on macOS Version 14.5
+ *
+ * Creates a custom driver table:
+ *    - Name: Name of the application
+ *    - Version: Version number of the application
+ *    - Path: Installation path of the application
+ *    - Architecture: Architecture type of the application
+ *    - Last modified: Last modified timestamp of the application
+ *    - Source: Source from where the application was obtained
+ *
+ **/
+
+// SSH command to retrieve retrieve OS information
+var sshConfig = {
+  'username': D.device.username(),
+  'password': D.device.password(),
+  'timeout': 10000
+}
+
+var applicationName = D.getParameter('applicationName')
+// Command to retrieve applications info
+var cmdApplications = 'system_profiler SPApplicationsDataType -json'
+// Fields to retrieve from the command output
+var fields = ' | grep -E "_name|version|path|arch_kind|obtained_from|lastModified"'
+var applications
+if (applicationName[0].toUpperCase() === 'ALL') {
```

Comment:
in both cases with applicationName or without, the command will be executed the it will pass throw the grep.
so without making things complicated make your filter on the json in javascript